### PR TITLE
Standardize ols

### PR DESCRIPTION
**Pull request by DavidGmidd**
Working on sample slopes standardization

### CODE REVIEW

**Comment by ethansaxenian** [comment]
Just checking - should I wait to review this since it will probably change after our meeting today?

File: src/components/SampleDistributionOLSEstimators/StandardNormalOLS.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import { dataFromDistribution } from '../../lib/stats-utils';
+import { sqrt } from 'mathjs';
+
+BellCurve(Highcharts);
+
+export default function StandardNormalOLS({ samples, sampleSize}) {
+  const [population] = useState(
+    dataFromDistribution('Normal', 2000, { mean: 0, standardDev: 1 })
```
Comment (ethansaxenian) [inline src/components/SampleDistributionOLSEstimators/StandardNormalOLS.js:16]: Why is a new population being generated here? We should be plotting the samples.

File: src/components/SampleDistributionOLSEstimators/StandardNormalOLS.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          baseSeries: 1,
+          zIndex: -1,
+          enableMouseTracking: false,
+          label: false,
+          showInLegend: false
+        },
+        {
+          name: 'Data',
+          type: 'scatter',
+          data: population.map(({ x }) => x),
```
Comment (ethansaxenian) [inline src/components/SampleDistributionOLSEstimators/StandardNormalOLS.js:73]: Ooohhh I see, for the bell curve! Ignore my previous comment.

File: src/components/SampleDistributionOLSEstimators/SlopeDistributionPlot.js [hunk 3]
```diff
@@ -3,22 +3,37 @@ import { max, min } from 'mathjs';
...
+  title="Distribution of Sample Slopes"
+  xMin={min(-5, ...plotData.map(({ x }) => x))}
+  xMax={max(5, ...plotData.map(({ x }) => x))}
+  yMax={max(4, ...plotData.map(({ y }) => y))}
+  xLabel="Slope"
+/> 
+
+if (standardized) {
+  plot = <StandardNormalOLS samples={samples}
+  sampleSize={+samples || 1} />
```
Comment (ethansaxenian) [inline src/components/SampleDistributionOLSEstimators/SlopeDistributionPlot.js:25]: this doesn't make sense, `samples` is a list

File: src/components/SampleDistributionOLSEstimators/SlopeDistributionPlot.js [hunk 4]
```diff
@@ -3,22 +3,37 @@ import { max, min } from 'mathjs';
...
 
-export default function SlopeDistributionPlot({ samples, populationShape }) {
+export default function SlopeDistributionPlot({ samples }) {
   const plotData = getCounts(samples.map(({ slope }) => slope));
+  const [standardized, setStandardized] = useState(false);
+
+  let plot = <DotPlot
+  series={[{ name: 'slopes', data: plotData }]}
+  title="Distribution of Sample Slopes"
+  xMin={min(-5, ...plotData.map(({ x }) => x))}
```
Comment (ethansaxenian) [inline src/components/SampleDistributionOLSEstimators/SlopeDistributionPlot.js:17]: Why were `xMin` and `xMax` changed here?

**Review by ethansaxenian — CHANGES_REQUESTED** [review]

File: src/components/SampleDistributionOLSEstimators/SlopeDistributionPlot.js (hunk 3 above)
Comment (DavidGmidd) [reply src/components/SampleDistributionOLSEstimators/SlopeDistributionPlot.js:25]: So we don't need sampleSize, right?

Comment (ethansaxenian) [reply src/components/SampleDistributionOLSEstimators/SlopeDistributionPlot.js:25]: Are you using it in `StandardNormalOLS`?

**Comment by DavidGmidd** [comment]
Is there a better way to calcualte the sd? @ethansaxenian 
`const` sdSlopes = `populationStandardDev(slopesArr)'

**Comment by ethansaxenian** [comment]
> Is there a better way to calcualte the sd? @ethansaxenian
> `const` sdSlopes = `populationStandardDev(slopesArr)'

Here is the code for `populationStandardDev`:

```
export const populationStandardDev = (popArray) => ((popArray.length > 0) ? std(popArray.map((p) => p.x)) : undefined)
```

This function tries to access `p.x`, where `p` is one object in the array. The sample objects in this case don't have an attribute `x`, so this particular function won't work, but you could write a similar one.

File: src/components/SampleDistributionOLSEstimators/StandardNormalOLS.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  let slopesArr = [];
+  for(var i in samples) {
+      totalSum = totalSum + samples[i].slope;
+      slopesArr.push(samples[i].slope);
+  }
+  var samplesArr = Object.keys(samples);
+  var totalLen = samplesArr.length;
+  const meanSlopes = totalSum / totalLen;
+
+  const sdSlopes = populationStandardDev(slopesArr)
```
Comment (ethansaxenian) [inline src/components/SampleDistributionOLSEstimators/StandardNormalOLS.js:57]: As I mentioned in an earlier comment, this function won't work here.

File: src/components/SampleDistributionOLSEstimators/StandardNormalOLS.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+  let totalSum = 0;
+  let slopesArr = [];
+  for(var i in samples) {
+      totalSum = totalSum + samples[i].slope;
+      slopesArr.push(samples[i].slope);
+  }
+  var samplesArr = Object.keys(samples);
+  var totalLen = samplesArr.length;
+  const meanSlopes = totalSum / totalLen;
```
Comment (ethansaxenian) [inline src/components/SampleDistributionOLSEstimators/StandardNormalOLS.js:55]: `mathjs` has a `mean` function that might be cleaner to use here.

**Review by ethansaxenian — CHANGES_REQUESTED** [review]

File: src/components/SampleDistributionOLSEstimators/SlopeDistributionPlot.js (hunk 4 above)
Comment (ethansaxenian) [reply src/components/SampleDistributionOLSEstimators/SlopeDistributionPlot.js:17]: I'm still confused by this

**Review by ethansaxenian — APPROVED** [review]

**Pull request merged**